nixos/modules/services/misc/mautrix-telegram.nix

```diff
@@ -0,0 +1,155 @@
+        #
+        #  Secret tokens should be specified using <option>environmentFile</option>
+        #  instead of this world-readable attribute set.
+        #'';
+      };
+
+      # TODO: uncomment this once mautrix-telegram v0.5.3 is out (https://github.com/tulir/mautrix-telegram/pull/332)
+      #environmentFile = mkOption {
+      #  type = types.nullOr types.path;
+      #  default = null;
+      #  description = ''
+      #    File containing environment variables to be passed to the mautrix-telegram service,
+      #    in which secret tokens can be specified securely by defining values for
+      #    <literal>MAUTRIX_TELEGRAM_TELEGRAM_API_ID</literal>,
+      #    <literal>MAUTRIX_TELEGRAM_TELEGRAM_API_HASH</literal> and optionally
+      #    <literal>MAUTRIX_TELEGRAM_TELEGRAM_BOT_TOKEN</literal>.
+      #  '';
+      #};
+
+      homeserverService = mkOption {
+        type = types.nullOr types.string;
+        default = "matrix-synapse.service";
+        description = ''
+          Matrix homeserver service to wait for when starting the application service.
+        '';
+      };
+    };
+  };
+
+  config = let
+    cfg = config.services.mautrix-telegram;
+    overrideConfigFile = pkgs.writeText "mautrix-telegram-config.json" (builtins.toJSON cfg.config);
+    serviceConfig = pkgs.runCommand "mautrix-telegram-config" { preferLocalBuild = true; } ''
+      mkdir -p "$out"
+
+      # the configuration file is rewritten to include the registration tokens
+      cp '${overrideConfigFile}' "$out/config.yaml"
+      chmod +w "$out/config.yaml"
+
+      ${pkgs.mautrix-telegram}/bin/mautrix-telegram \
+        --generate-registration \
+        --base-config='${pkgs.mautrix-telegram}/example-config.yaml' \
+        --config="$out/config.yaml" \
+        --registration="$out/registration.yaml"
+    '';
+  in mkIf cfg.enable {
+    systemd.services.mautrix-telegram = {
+      description = "Mautrix-Telegram, a Matrix-Telegram hybrid puppeting/relaybot bridge.";
+
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ] ++ optional (cfg.homeserverService != null) cfg.homeserverService;
+      after = [ "network-online.target" ] ++ optional (cfg.homeserverService != null) cfg.homeserverService;
+
+      preStart = ''
+        cd ${pkgs.mautrix-telegram}
+        ${pkgs.mautrix-telegram.alembic}/bin/alembic -x config='${serviceConfig}/config.yaml' upgrade head
+      '';
+
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = ''
+          ${pkgs.mautrix-telegram}/bin/mautrix-telegram \
+            --config='${serviceConfig}/config.yaml'
+        '';
+        Restart = "always";
+        DynamicUser = true;
+        StateDirectory = baseNameOf dataDir;
+        # TODO: uncomment this once mautrix-telegram v0.5.3 is out (https://github.com/tulir/mautrix-telegram/pull/332)
+        #EnvironmentFile = cfg.environmentFile;
```

Comment:
Can't we just apply the patch onto the current package?

Reply:
Or we can just wait for the next release.
It shouldn't be long given that [v0.6.0-rc1](https://github.com/tulir/mautrix-telegram/releases/tag/v0.6.0-rc1) has been published recently.

Reply:
#63947 I will update it once the stable release is out